packages/@expo/cli/src/utils/telemetry/index.ts

```diff
@@ -0,0 +1,28 @@
+import { DetachedClient } from './DetachedClient';
+import { RudderClient } from './RudderClient';
+import type { TelemetryClient, TelemetryEvent, TelemetryProperties } from './types';
+import { env } from '../env';
+
+const debug = require('debug')('expo:telemetry') as typeof console.log;
+
+function getClient(): TelemetryClient | null {
+  if (env.EXPO_NO_TELEMETRY) return null;
+
+  const client = env.EXPO_NO_TELEMETRY_DETACH
+    ? new RudderClient() // Block the CLI process when sending telemetry, useful for testing
+    : new DetachedClient(); // Do not block the CLI process when sending telemetry
+
+  process.once('SIGINT', () => client.flush());
+  process.once('SIGTERM', () => client.flush());
+  process.once('beforeExit', () => client.flush());
```

Comment:
@EvanBacon I noticed that the `installExitHooks` aren't triggered. Not on normal process exit, or when canceling with <kbd>ctrl</kbd> + <kbd>c</kbd>.

This works for now, but if you have a better idea, I'm all ears.